tests/test_bots.py

```diff
@@ -24,3 +31,77 @@ def setUp(self) -> None:
     def test_run(self) -> None:
         # TODO
         pass
+
+
+class RandMiniMaxBot(Bot):
+    """In the phase1, this bot plays random, and in the phase2, it plays minimax.
+    The opponent is random."""
+
+    def __init__(self) -> None:
+        self.bot_phase1 = RandBot(42)
+        self.bot_phase2 = MiniMaxBot()
+
+    def get_move(self, state: PlayerPerspective, leader_move: Optional[Move]) -> Move:
+        if state.get_phase() == GamePhase.ONE:
+            return self.bot_phase1.get_move(state, leader_move)
+        elif state.get_phase() == GamePhase.TWO:
+            return self.bot_phase2.get_move(state, leader_move)
+        else:
+            raise ValueError("Phase ain't right.")
+
+
+class RdeepMiniMaxBot(Bot):
+    """In the phase1, this bot plays random, and in the phase2, it plays minimax.
+    The opponent is random."""
+
+    def __init__(self) -> None:
+        self.bot_phase1 = RdeepBot(num_samples=16, depth=4, rand=random.Random(42))
+        self.bot_phase2 = MiniMaxBot()
+
+    def get_move(self, state: PlayerPerspective, leader_move: Optional[Move]) -> Move:
+        if state.get_phase() == GamePhase.ONE:
+            return self.bot_phase1.get_move(state, leader_move)
+        elif state.get_phase() == GamePhase.TWO:
+            return self.bot_phase2.get_move(state, leader_move)
+        else:
+            raise ValueError("Phase ain't right.")
+
```

Comment:
@tae898 We would need these unit tsts to be much more fine grained. Basically, we should only have this Bot and call methods (get_move) on it directly. We then provide it with the needed arguments and checks whether it is giving us the move we expect.